pkg/api/validate_plugin_test.go

```diff
@@ -0,0 +1,345 @@
+   genevaMetrics:
+     key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlCT1FJQkFBSkJBTko2cWhjWmlBK0tsUURETlZqQTY0TVRJbSt3WGFWUnZ6Q2Zwbm9ya0Y0OVJHMVYvVm5mClZCTmVPNTBvb3E1ZGNpcElOM284bmVwY09QQU5Ybk5vVkVNQ0F3RUFBUUpBSFNIclR2MHlydXdBaWJWN09jaWkKRUdkaW1kRHdkVVJtVVNXWDFrc1hWV09uTXFxeFk4c1ZEZTQrOVNqbW1uMHRpZjc3UDRHWE0zUWxKSjFXa0tvQQo4UUloQVBPWjhjRDd0NTNBazIzOWh1bytMR1FnNUZZaVdVM0JGWTJ1VUQ0RG1EL0xBaUVBM1RFbHdFcC8ybXN5CkVlaXNlc3B6ZlBqQXVSME16clRoS3FEcTEwa3BQbWtDSUdFaThORElUd2FicE81R0cwZEt0WDdUMHRrNTV5eG4KSXdZVkRUQTlWTGVUQWlBd2dhcXB0S3k5Rld6eGlIanFwS01XOE9ZeXNqQXcxSEhjaTFWMHlOS0dvUUlnWlZiVQpNZU1kQVdVdkVJbXowY0RnQ3BLTCtqNDAySm1iZFZ1dkhNNyt3QVU9Ci0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg==
+     cert: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUI2akNDQVpTZ0F3SUJBZ0lKQVBaMC8ydDJqYXJ5TUEwR0NTcUdTSWIzRFFFQkN3VUFNQlV4RXpBUkJnTlYKQkFNVENtdDFZbVZ5Ym1WMFpYTXdIaGNOTVRneE1URXlNVGN5TWpJeFdoY05NakF4TVRFeU1UY3lNakl4V2pBVgpNUk13RVFZRFZRUURFd3ByZFdKbGNtNWxkR1Z6TUZ3d0RRWUpLb1pJaHZjTkFRRUJCUUFEU3dBd1NBSkJBTko2CnFoY1ppQStLbFFERE5WakE2NE1USW0rd1hhVlJ2ekNmcG5vcmtGNDlSRzFWL1ZuZlZCTmVPNTBvb3E1ZGNpcEkKTjNvOG5lcGNPUEFOWG5Ob1ZFTUNBd0VBQWFPQnhqQ0J3ekFPQmdOVkhROEJBZjhFQkFNQ0JhQXdEQVlEVlIwVApBUUgvQkFJd0FEQ0JvZ1lEVlIwUkJJR2FNSUdYZ2dwcmRXSmxjbTVsZEdWemdnMXRZWE4wWlhJdE1EQXdNREF3CmdnMXRZWE4wWlhJdE1EQXdNREF4Z2cxdFlYTjBaWEl0TURBd01EQXlnZ3ByZFdKbGNtNWxkR1Z6Z2hKcmRXSmwKY201bGRHVnpMbVJsWm1GMWJIU0NGbXQxWW1WeWJtVjBaWE11WkdWbVlYVnNkQzV6ZG1PQ0pHdDFZbVZ5Ym1WMApaWE11WkdWbVlYVnNkQzV6ZG1NdVkyeDFjM1JsY2k1c2IyTmhiREFOQmdrcWhraUc5dzBCQVFzRkFBTkJBTkF2CjUwaXJlRHRqUkRiMVZydjJEcmRGZXhrT2hhZzNJM3dEVWlPb0loYjVuTkNNRjdnMEF1VEpOWkhFUXFOWDNYb1gKYkZpRkExdUxLZEZMc1B1T1dTUT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
+`)
+
+func TestPluginTemplateValidate(t *testing.T) {
+	tests := map[string]struct {
+		f            func(*pluginapi.Config)
+		expectedErrs []error
+	}{
+		"test yaml parsing": { // test yaml parsing
+
+		},
+		"invalid ImageOffer": {
+			f: func(t *pluginapi.Config) {
+				t.ImageOffer = "non-osa"
+			},
+			expectedErrs: []error{errors.New(`imageOffer should be osa`)},
+		},
+		"invalid ImagePublisher": {
+			f: func(t *pluginapi.Config) {
+				t.ImagePublisher = "microsoft"
+			},
+			expectedErrs: []error{errors.New(`imagePublisher should be redhat`)},
+		},
+		"invalid ImageSKU": {
+			f: func(t *pluginapi.Config) {
+				t.ImageSKU = "osa_310"
+			},
+			expectedErrs: []error{errors.New(`invalid ImageSKU "osa_310"`)},
+		},
+		"invalid ImageVersion": {
+			f: func(t *pluginapi.Config) {
+				t.ImageVersion = "1.23.456"
+			},
+			expectedErrs: []error{errors.New(`invalid ImageVersion "1.23.456"`)},
+		},
+		"empty GenevaLoggingSector": {
```

Comment:
try separating this to a different test / part of a different test - validate an empty file and check the list of errors returned once?